test/coupon_code_test.rb

```diff
@@ -59,6 +59,10 @@ def generate(*args)
       CouponCode.validate('CTFM-1K7Q', 2).must_be_nil
     end
 
+    it 'handles code replaced zero with letter O' do
```

Comment:
I think this should also be updated too. Such as 'should reject a short code contains any invalid character.'